hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java

```diff
@@ -76,14 +75,20 @@ public ConsensusSubmitMessageHandler() {
         // Exists for injection
     }
 
+    @Override
+    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
+        final ConsensusSubmitMessageTransactionBody op = txn.consensusSubmitMessageOrThrow();
+        validateTruePreCheck(op.hasTopicID(), INVALID_TOPIC_ID);
+        validateFalsePreCheck(op.message().length() == 0, INVALID_TOPIC_MESSAGE);
```

Comment:
Slight tweak:
```suggestion
        validateFalsePreCheck(op.message().length() > 0, INVALID_TOPIC_MESSAGE);
```

Reply:
validateFalsePreCheck() checks condition true, then throws the exception, here we want to check length 0 message. I was also wondering why we didn't name it validateTruePreCheck(). There is a unit test, failsIfMessageIsEmpty() in ConsensusSubmitMessageTest.

Reply:
At least `>= 0`, we wouldn't want a value of `-1` to pass. Though...I admit it seems highly unlikely

Reply:
validateFalsePreCheck(op.message().length() > 0, INVALID_TOPIC_MESSAGE); means msg length>0 will throw INVALID_TOPIC_MESSAGE. That's wrong.

Reply:
Sorry, I meant 
`validateTruePreCheck(op.message().length() >= 0, INVALID_TOPIC_MESSAGE)`

Reply:
The length of op.message() won't be < 0. We just need to check it's not 0 here since > 0 is valid.